index.d.ts

```diff
@@ -0,0 +1,78 @@
+export enum ZigbeeDeviceAccess {
+    STATE = 1,
+    SET = 2,
+    STATE_SET = 3,
+    STATE_GET = 5,
+    ALL = 7,
+}
+
+export interface ZigbeeDeviceExposeGeneric {
+    access: ZigbeeDeviceAccess
+    description: string
+    name: string
+    property: string
+    type: string
+}
+
+export interface ZigbeeDeviceExposeEnum {
+    type: 'enum'
+    values: string[]
+}
+
+export interface ZigbeeDeviceExposeNumeric {
+    type: 'numeric'
+    value_min: number
+    value_max: number
+}
+
+export interface ZigbeeDeviceExposeBinary {
+    type: 'binary'
+    value_off: boolean
+    value_on: boolean
+}
+
+export interface ZigbeeDeviceExposeText {
+    type: 'text'
+}
+
+export interface ZigbeeDeviceExposeLight {
+    type: 'light'
+    features: ZigbeeDeviceExpose[]
+}
+
+export interface ZigbeeDeviceExposeComposite {
+    type: 'composite'
+    features: ZigbeeDeviceExpose[]
+}
+
+export type ZigbeeDeviceExpose = ZigbeeDeviceExposeGeneric & (ZigbeeDeviceExposeEnum | ZigbeeDeviceExposeNumeric | ZigbeeDeviceExposeBinary | ZigbeeDeviceExposeText | ZigbeeDeviceExposeLight | ZigbeeDeviceExposeComposite)
+
+export interface ZigbeeDevice {
```

Comment:
Device definition is slightly more complicated
https://github.com/nurikk/z2m-frontend/blob/a341b4dd5704d645c11845deb2213915119a0e03/src/types.ts#L195

Reply:
Wouldn't it make sense to copy the entire definition then? Or move them to a shared library?

Reply:
Yes, having separate shared library will be nice and handy, I could start using it as well

Reply:
For [my zigbee2mqtt homebridge plugin](https://github.com/itavero/homebridge-z2m), I've also declared a lot of these models in my own code (a bit less detailed though, as I don't need all the information in my plugin).
For me having a separate library that defines these interfaces (which, as far as I can tell, map to the data on MQTT as well), could also be very nice.